src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1591,6 +1592,41 @@ public string ToString([StringSyntax(StringSyntaxAttribute.NumericFormat)] strin
             return BigNumber.FormatBigInteger(this, format, NumberFormatInfo.GetInstance(provider));
         }
 
+        private string GetDebuggerDisplay()
+        {
+            // For very big numbers, ToString can be too long or even timeout for Visual Studio to display
+            // Display an fast estimation value instead
+
+            // Use ToString for small values
+
+            if (_bits is null || _bits.Length <= 2)
```

Comment:
```suggestion
            if (_bits is null || _bits.Length <= 15)
```
2 is too small?

Reply:
Each element in `_bits` is 32 bits. Starting at 64 bits doesn't look so small.